packages/node/src/handlers.ts

```diff
@@ -392,10 +407,36 @@ export function requestHandler(
     local.add(req);
     local.add(res);
     local.on('error', next);
+
     local.run(() => {
-      getCurrentHub().configureScope(scope =>
-        scope.addEventProcessor((event: Event) => parseRequest(event, req, options)),
-      );
+      const currentHub = getCurrentHub();
+
+      currentHub.configureScope(scope => {
+        scope.addEventProcessor((event: Event) => parseRequest(event, req, options));
+        const client = currentHub.getClient<NodeClient>();
+        if (isAutoSessionTrackingEnabled(client)) {
+          const scope = currentHub.getScope();
+          if (scope) {
+            // Set `status` of `_requestSession` to Ok, at the beginning of the request
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+            (scope as any)._requestSession = { status: RequestSessionStatus.Ok };
+          }
+        }
+      });
+
+      res.once('finish', () => {
+        const client = currentHub.getClient<NodeClient>();
+        if (isAutoSessionTrackingEnabled(client)) {
+          setImmediate(() => {
+            if (client) {
+              // Calling _captureRequestSession to capture request session at the end of the request by incrementing
+              // the correct SessionAggregates bucket i.e. crashed, errored or exited
+              // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+              (client as any)._captureRequestSession();
+            }
+          });
+        }
```

Comment:
This logic all boils down to the client existence check. If it's not present, there's no need to schedule anything or check for tracking.

```suggestion
        const client = currentHub.getClient<NodeClient>();
        
        if (!client || !isAutoSessionTrackingEnabled(client)) {
          setImmediate(() => {
            // Calling _captureRequestSession to capture request session at the end of the request by incrementing
            // the correct SessionAggregates bucket i.e. crashed, errored or exited
            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
            (client as any)._captureRequestSession();
          });
        }
```